Fixing issue #153 (failing tests)

mpatsute opened the pull request:
cc @bajtos @raymondfeng @ritch @superkhau
Hey All, "npm test" was failing, I got it working after some fixes. Had to make a change in SwaggerRouter.ts and added some test related dependencies in package.json.
Please take a look, and merge, if everything looks ok to you.

superkhau commented on package.json:
```diff
@@ -18,7 +18,12 @@
+    "path-to-regexp": "^1.7.0",
+    "should": "^11.2.1",
+    "sinon": "^2.1.0",
+    "supertest": "^3.0.0"
```
this looks suspicious -- all the test deps should be inside packages/testlab

superkhau commented on packages/loopback/lib/router/SwaggerRouter.ts:
```diff
@@ -197,8 +197,8 @@ class Endpoint {
   pathParams: {[key: string]: string}): any[] {
-  const args = [];
-  
+  const args = new Array<string>();
```
this looks suspicious too -- does `const args: Array<string> = []` work?

raymondfeng replied:
You can use `const args: string[] = [];` or `const args = [] as string[]`;

superkhau replied:
I good with both of your suggestions (or mine).

mpatsute commented:
Fixed the PR based on review comments:
1. Removed newly added dependencies in the top level package.json
2. Fixed SwaggerRouter.ts, to declare type properly

bajtos commented:
Landed, thank you @mpatsute for taking the initiative to fix this issue!
